src/Microsoft.AspNet.Mvc.Razor.Host/Directives/DefaultCodeTreeCache.cs

```diff
@@ -43,22 +43,24 @@ public DefaultCodeTreeCache(IFileProvider fileProvider)
         public CodeTree GetOrAdd([NotNull] string pagePath,
                                  [NotNull] Func<IFileInfo, CodeTree> getCodeTree)
         {
-            return _codeTreeCache.GetOrSet(pagePath, getCodeTree, OnCacheMiss);
-        }
+            CodeTree codeTree;
+            if (!_codeTreeCache.TryGetValue(pagePath, out codeTree))
+            {
+                // GetOrAdd is invoked for each _GlobalImport that might potentially exist in the path.
```

Comment:
yeah, right... @dougbu  mentioned it to me...i was planning to revert this change as his PR is going to have this change...